ros2_batch_job/__main__.py

```diff
@@ -579,23 +562,37 @@ def run(args, build_function, blacklisted_package_names=None):
                 exit_on_error=False)
             print('# END SUBSECTION')
 
-        if not args.connext:
+        # blacklist rmw packages as well as their dependencies where possible
+        if 'rmw_connext_cpp' in args.ignore_rmw:
             blacklisted_package_names += [
-                'connext_cmake_module',
                 'rmw_connext_cpp',
-                'rmw_connext_dynamic_cpp',
-                'rmw_connext_shared_cpp',
                 'rosidl_typesupport_connext_c',
                 'rosidl_typesupport_connext_cpp',
             ]
-        if not args.fastrtps:
+        if 'rmw_connext_dynamic_cpp' in args.ignore_rmw:
+            blacklisted_package_names += [
+                'rmw_connext_dynamic_cpp',
+            ]
+        if 'rmw_connext_cpp' in args.ignore_rmw and 'rmw_connext_dynamic_cpp' in args.ignore_rmw:
+            blacklisted_package_names += [
+                'connext_cmake_module',
+                'rmw_connext_shared_cpp',
+            ]
```

Comment:
If both of the two RMW packages using Connext are ignored the supporting packages can be ignored too.

Similar for FastRTPS and OpenSplice below.